### PR TITLE
Allow automatically initiating corresponding protocol session negotiation for all configured devices

### PR DESCRIPTION
**Pull request by ashutshkumr**
Fixes https://github.com/open-traffic-generator/models/issues/270. See the documentation [here](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/open-traffic-generator/models/auto-start/artifacts/openapi.yaml&nocors).

A new global option for devices has been introduced in `config` to indicate whether all underlying resources for configured devices shall be created and corresponding protocol session negotiation shall be negotiated upon `set_config` API call. Corresponding gosnappi snippet would look like so:

   ```go
   // do not initiate protocol session negotiation upon set_config API call
   config.Options().DeviceOptions().SetAutoStartAll(false)
   ```

Default is `true`.

### CODE REVIEW

**Comment by bortok** [comment]
With auto_start_all being set to true by default, have we analyzed an impact it would have on existing scripts that would start protocols explicitly?

**Comment by bortok** [comment]
Second comment - it would be quite useful with this change to review if we have documentation on otg control calls, what should be a typical sequence, importance to wait for convergence and so on. I'm open to contribute to the such doc.

**Review by apratimmukherjee — APPROVED** [review]
Looks good to me.

**Comment by ashutshkumr** [comment]
> 

@bortok The impact on existing tests would be that the StartProtocol call would return a warning mentioning that protocols are already started when auto_start=true. Hence no breaking changes are required in tests.

**Comment by ashutshkumr** [comment]
> Second comment - it would be quite useful with this change to review if we have documentation on otg control calls, what should be a typical sequence, importance to wait for convergence and so on. I'm open to contribute to the such doc.

@bortok thanks for the suggestion, I've opened an issue to track changes regarding this: https://github.com/open-traffic-generator/models/issues/288

File: device/device.yaml [hunk 1]
```diff
@@ -63,3 +63,18 @@ components:
           x-include: ../common/common.yaml#/components/schemas/Named.Object/properties/name
           x-field-uid: 7
       required: [name]
+    Device.Options:
+      description: >-
+        Common options that apply to all configured devices. 
+      type: object
+      properties:
+        auto_start_all:
+          description: >-
```
[the remark this answers is not in the document]
Comment (ashutshkumr) [reply device/device.yaml:72]: added more description for set_protocol_state

**Pull request merged**